### ISSUE TITLE
Wrong Bus for `play_stream` in Web Export Only (in 4.3, worked in 4.2)

### ISSUE THREAD

**Issue by TinyTakinTeller** [godotengine/godot#100696]
### Tested versions

- Reproducible in: v4.3.stable.official [77dcf97d8]
- Not reproducible in: v4.2.stable.official [46dc27791]

### System information

Windows 11 - Google Chrome

### Issue description

The method `play_stream` has a new argument `bus` in 4.3.
- 4.3. https://docs.godotengine.org/en/4.3/classes/class_audiostreamplaybackpolyphonic.html
- 4.2. https://docs.godotengine.org/en/4.2/classes/class_audiostreamplaybackpolyphonic.html

When testing DESKTOP (run), the same code works in both 4.2 and 4.3 versions.

When testing WEB export, the same code works in 4.2 but does not work in 4.3. version.
_(Uses Master bus instead of set e.g. "Music" bus.)_
_(Note that in web export, you need to click on the project window to grab focus to allow creation of audio context.)_

### Steps to reproduce

The minimal code below: 
- in 4.3. it is muted in desktop but playing in web
- in 4.2. it is muted in both desktop and web

```
extends Node2D

func _ready():
	AudioServer.set_bus_mute(AudioServer.get_bus_index("Music"), true)
	
	var audio = load("res://menu_doodle_2_loop.ogg")
	
	var player = AudioStreamPlayer.new()
	var stream = AudioStreamPolyphonic.new()

	player.stream = stream
	player.process_mode = process_mode
	player.bus = "Music"

	player.stream.polyphony = 1
	player.max_polyphony = 1

	add_child(player)

	player.play()

	var playback = player.get_stream_playback() as AudioStreamPlaybackPolyphonic

	var stream_id = playback.play_stream(audio as AudioStream)
```

Temporary workaround (in GDScript) for 4.3. web export is to set bus explicitly in the new argument (pass it `player.bus`).

But this is not compatible with 4.2. or earlier because the new argument exists only in newer versions.

### Minimal reproduction project (MRP)

[new-project-4_3.zip](https://github.com/user-attachments/files/18218458/new-project-4_3.zip)
[new-project-4_2.zip](https://github.com/user-attachments/files/18218459/new-project-4_2.zip)

**Comment by 3lis4** [comment]
This issue still exists in 4.4-stable.

I've made another MRP for 4.4-stable. The AudioStreamPlayer bus is "PitchBus", which makes it sound "higher".

[MRP_Bus_changes_on_web_exports.zip](https://github.com/user-attachments/files/19232838/MRP_Bus_changes_on_web_exports.zip)

Steps to reproduce :
- if you launch the game in editor, you'll hear the music with a high pitch,
- if you launch the game on a local server using Remote Deploy > Run in browser, you won't hear the high pitch, but you'll hear the normal pitch instead, probably because AudioStreamPlayer plays on Master bus instead of PitchBus.

@adamscott I think this issue might interest you as you've worked on similar issues. (Please tell me if I shouldn't ping you).

**Comment by adamscott** [comment]
> This issue still exists in 4.4-stable.
> 
> I've made another MRP for 4.4-stable. The AudioStreamPlayer bus is "PitchBus", which makes it sound "higher".
> 
> [MRP_Bus_changes_on_web_exports.zip](https://github.com/user-attachments/files/19232838/MRP_Bus_changes_on_web_exports.zip)
> 
> Steps to reproduce :
> 
>     * if you launch the game in editor, you'll hear the music with a high pitch,
> 
>     * if you launch the game on a local server using Remote Deploy > Run in browser, you won't hear the high pitch, but you'll hear the normal pitch instead, probably because AudioStreamPlayer plays on Master bus instead of PitchBus.
> 
> 
> [@adamscott](https://github.com/adamscott) I think this issue might interest you as you've worked on similar issues. (Please tell me if I shouldn't ping you).

The behavior you're experiencing, @3lis4, is intended. Please read [the blog post I wrote on the subject](https://godotengine.org/article/progress-report-web-export-in-4-3/#sample-playback-to-the-rescue). By default, starting from 4.3, the Web platform uses samples instead of streams. Unfortunately, for now, bus effects aren't yet supported.

**Comment by adamscott** [comment]
@TinyTakinTeller I don't understand the link between the new argument with the issue, though.

**Comment by TinyTakinTeller** [comment]
> [@TinyTakinTeller](https://github.com/TinyTakinTeller) I don't understand the link between the new argument with the issue, though.

Yeah I'm not sure, perhaps a coincidence it was added then and there - but seems it can be used as a workaround for now, as mentioned.

**Comment by adamscott** [comment]
Closing the issue then!

**Comment by TinyTakinTeller** [comment]
> Closing the issue then!

Oh, but shouldn't the web and desktop version act the same when calling `.play_stream` function ?

(Instead of one the player's bus and the other using master bus, if no bus is provided.)

**Comment by adamscott** [comment]
I'll take a look again.

**Comment by TinyTakinTeller** [comment]
> [@TinyTakinTeller](https://github.com/TinyTakinTeller) I don't understand the link between the new argument with the issue, though.

Sorry If I misspoke earlier, the relevance of the new arguments is, that if it is left empty, then behaves differently in web and desktop exports (one uses master bus, other uses bus that was set in the player) in Godot 4.3. (there is no such argument in 4.2. so there we do not have such inconsistency.)

**Comment by adamscott** [comment]
Just realized that I introduced that change. But I don't remember why exactly. And that's unfortunate.

Technically, if everything works as intended, it's the desktop exports that aren't working properly, as `play_stream` defaults to `&"Master"`.
